crew

```diff
@@ -209,30 +239,26 @@ end
 def resolveDependencies
   @dependencies = []
 
-  def pushDeps
-    if @pkg.dependencies
-      @dependencies.unshift @pkg.dependencies
-      
-      @pkg.dependencies.each do |dep|
-        search dep, true
-        pushDeps
+  if @pkg.dependencies
+    @pkg.dependencies.each do |dep|
+      if not search(dep, true)
+        if not @dependencies.include? dep
+          @dependencies.push(dep)
+        end
       end
     end
   end
-  
-  pushDeps
```

Comment:
@jam7: I refactored pushDeps so it recursively calls itself to drill down and get every possible dependency.  Only dependencies that have not been installed yet will be prompted.